src/training/train.cpp

```diff
@@ -157,27 +171,85 @@ void TestInference(RecordInput* input, GraphInference* inference) {
 
 }
 
-void Train(RecordInput* input, GraphInference* inference) {
-  inference->SSVMInit(FLAGS_regularization_const, FLAGS_svm_margin);
+void Train(RecordInput* input, GraphInference* inference, int fold_id) {
```

Comment:
It is not about the initialization, but about the loop in Train. This was a small method that grew substantially with a lot of ifs you add and these ifs are hard to read.